lib/cache.js

```diff
@@ -6,13 +6,12 @@ const async = require('async')
 var Cache = module.exports = function (trie) {
   this._cache = Tree()
   this._checkpoints = []
-  this._deletes = []
```

Comment:
This refactoring seems to make very much sense, to remove this separate delete stack and put it in the values itself.